pkg/telemetry/telemetry.go

```diff
@@ -0,0 +1,246 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/jackc/pgtype"
+
+	"github.com/prometheus/client_golang/prometheus"
+	io_prometheus_client "github.com/prometheus/client_model/go"
+
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/pgmodel/common/schema"
+	"github.com/timescale/promscale/pkg/pgmodel/model/pgutf8str"
+	"github.com/timescale/promscale/pkg/pgxconn"
+)
+
+// ExecPlatform To fill this variable in build time, use linker flags.
+// Example: go build -ldflags="-X github.com/timescale/promscale/pkg/telemetry.ExecPlatform=<any_string>" ./cmd/promscale/
```

Comment:
I gave an attempt and didn't seem straightforward. Will need help there. Plus, there seems like we may need to set github env as well.

Reply:
I think you can set ldflags in the build section ([docs](https://goreleaser.com/customization/build/)) so for us it would be around [line 16 ](https://github.com/timescale/promscale/blob/master/build/.goreleaser.yml#L16) and it would be 
```
ldflags:
 - -s -w -X main.version={{.Version}} -X main.commit={{.Commit}} -X main.date={{.Date}} -X main.builtBy=goreleaser -X github.com/timescale/promscale/pkg/telemetry.ExecPlatform=goreleaser
```

Also isn't a better name here BuildPlatform not ExecPlatform? This will indicate goreleaser vs apt vs rpm etc right?

Reply:
That approach for goreleaser should work. But I am confused here on some points:
1. Does the current master does not implement commit & version information?
2. I thought goreleaser will release for apt & rpm too, not on individual platforms

Reply:
Lets discuss this on call

Reply:
`binary, deb, rpm` in ExecPlatform (replaced by BuildPlatform)